src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,222 @@
+        }
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getQuery = "SELECT c.id AS car_id, model, registration_number, "
+                + "m.id AS manufacturer_id, m.name, m.country "
+                + "FROM cars c JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.id = ? AND c.is_deleted = FALSE";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getStatement = connection.prepareStatement(getQuery)) {
+            getStatement.setLong(1, id);
+            ResultSet resultSet = getStatement.executeQuery();
+            if (resultSet.next()) {
+                car = parseCar(resultSet);
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't get car by id = " + id, throwable);
+        }
+        if (car != null) {
+            car.setDrivers(getAllDriversByCarId(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllQuery = "SELECT c.id AS car_id, model, registration_number, "
+                + "m.id AS manufacturer_id, m.name, m.country "
+                + "FROM cars c JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = FALSE";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getAllStatement = connection.prepareStatement(getAllQuery)) {
+            ResultSet resultSet = getAllStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseCar(resultSet));
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't get list of cars!", throwable);
+        }
+        cars.forEach(car -> car.setDrivers(getAllDriversByCarId(car.getId())));
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateQuery = "UPDATE cars SET model = ?, registration_number = ?, "
+                + "manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        int updateRecordsNumber;
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement updateStatement =
+                         connection.prepareStatement(updateQuery)) {
+            updateStatement.setString(1, car.getModel());
+            updateStatement.setString(2, car.getRegistrationNumber());
+            updateStatement.setLong(3, car.getManufacturer().getId());
+            updateStatement.setLong(4, car.getId());
+            updateRecordsNumber = updateStatement.executeUpdate();
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't update car: " + car, throwable);
+        }
+        if (updateRecordsNumber > 0) {
```

Comment:
this check cant do harm. What if the only change is removal of driver, in this case updateQuery won't change any lines

Reply:
I agree. I was very reinsured in both cases)))